destination/destination.go

```diff
@@ -0,0 +1,245 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/conduitio-labs/conduit-connector-dropbox/pkg/dropbox"
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	config Config
+	client dropbox.Client
+	// sessions stores the active sessions for chunk uploads
+	session *Session
+}
+
+func NewDestination() sdk.Destination {
+	return sdk.DestinationWithMiddleware(&Destination{})
+}
+
+func (d *Destination) Config() sdk.DestinationConfig {
+	return &d.config
+}
+
+func (d *Destination) Open(ctx context.Context) error {
+	sdk.Logger(ctx).Info().Msg("Opening Dropbox destination")
+	var err error
+
+	if d.client == nil {
+		d.client, err = dropbox.NewHTTPClient(d.config.Token, 0)
+		if err != nil {
+			return fmt.Errorf("error creating dropbox client: %w", err)
+		}
+	}
+	d.session = NewSession()
+	return nil
+}
+
+func (d *Destination) Write(ctx context.Context, records []opencdc.Record) (int, error) {
+	for i, record := range records {
+		//nolint:exhaustive // default handles all non-delete operations
+		switch record.Operation {
+		case opencdc.OperationDelete:
+			filepath, ok := record.Metadata["file_path"]
+			if !ok {
+				return i, ErrMissingFilePath
+			}
+			err := d.client.DeleteFile(ctx, filepath)
+			if err != nil {
+				return i, fmt.Errorf("error deleting file: %w", err)
+			}
+
+		default:
+			chunked, ok := record.Metadata["is_chunked"]
+			if ok && chunked == "true" {
```

Comment:
Nitpick: can be simplified with `case: record.Metadata["is_chunked"] == "true"`.